### PR TITLE
Enforced uuid validation

### PR DESCRIPTION
**Pull request by HeahDude**

### CODE REVIEW

File: web/app.php [hunk 1]
```diff
@@ -21,7 +21,6 @@
 
 /** @var \Composer\Autoload\ClassLoader $loader */
 $loader = require __DIR__.'/../app/autoload.php';
-include_once __DIR__.'/../var/bootstrap.php.cache';
```
Comment (tgalopin) [inline web/app.php]: Why?

File: composer.json [hunk 2]
```diff
@@ -73,11 +73,9 @@
     "scripts": {
         "symfony-scripts": [
             "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
```
Comment (HeahDude) [inline composer.json]: https://github.com/symfony/symfony-standard/blob/master/web/app.php#L8, opcache does the job, it adds overhead for no gain.

File: web/app.php (hunk 1 above)
Comment (HeahDude) [inline web/app.php]: Same as above.

File: composer.json (hunk 2 above)
Comment (tgalopin) [inline composer.json]: Nice! However, I would prefer to have a different PR, it's not related to the current one.

Comment (HeahDude) [inline composer.json]: ok, not a problem.

Comment (HeahDude) [inline composer.json]: Done.

File: src/AppBundle/Controller/DonationController.php [hunk 3]
```diff
@@ -91,11 +92,15 @@ public function callbackAction(Request $request)
     {
         $id = explode('_', $request->query->get('id'))[0];
```
Comment (hhamon) [inline src/AppBundle/Controller/DonationController.php]: This line and the following condition should be encapsulated somewhere in a static method. Something similar to this.

```
try {
    Donation::validateId($request->query->get('id'));
} catch (InvalidDonationIdException $e) {
    return $this->redirectToRoute('donation_index');
}

File: tests/AppBundle/Controller/AssetsControllerTest.php [hunk 4]
```diff
@@ -91,7 +91,7 @@ public function testStaticMaps()
 
     public function testStaticMapsWithWrongQuery()
     {
-        $client = static::createClient();
+        $client = $this->client;
```
Comment (hhamon) [inline tests/AppBundle/Controller/AssetsControllerTest.php]: Why making a copy?

File: tests/AppBundle/Controller/AssetsControllerTest.php [hunk 5]
```diff
@@ -45,7 +45,7 @@ public function testInvalidAssetWithSignatureIsNotFound()
 
     public function testStaticMaps()
     {
-        $client = static::createClient();
+        $client = $this->client;
```
Comment (hhamon) [inline tests/AppBundle/Controller/AssetsControllerTest.php]: Why making a copy?

**Comment by hhamon** [comment]
Can't we manage the UuidException in a custom `kernel.exception` listener? We could then simplify the controllers and avoid all these try/catch blocks.

File: src/AppBundle/Controller/DonationController.php (hunk 3 above)
Comment (HeahDude) [inline src/AppBundle/Controller/DonationController.php]: The donation controller needs more love but it's for another PR ;).

File: tests/AppBundle/Controller/AssetsControllerTest.php (hunk 5 above)
Comment (HeahDude) [inline tests/AppBundle/Controller/AssetsControllerTest.php]: IT was a quick fix. Addressed.

File: tests/AppBundle/Controller/AssetsControllerTest.php (hunk 4 above)
Comment (HeahDude) [inline tests/AppBundle/Controller/AssetsControllerTest.php]: Fixed.

**Comment by HeahDude** [comment]
>Can't we manage the UuidException in a custom kernel.exception listener? We could then simplify the controllers and avoid all these try/catch blocks.

I'm not sure about the exception listener, there is already one in charge for the response because we throw an HttpExceptionInterface, this could not be done in a listener.

Repository have to throw an InvalidArgumentException, letting the controller transform it to an HTTP one make it explicit also when we read the code, and allow to adjust handling things per actions.

We already have other try catch/blocks in controllers so it looks consistent to me, especially the Facebook one.

@tgalopin what do you think?

File: tests/AppBundle/Controller/AdherentControllerTest.php [hunk 6]
```diff
@@ -18,7 +18,7 @@
 use Tests\AppBundle\MysqlWebTestCase;
 
 /**
- * @group functional
+ * @group functionnal
```
Comment (tony-tran) [inline tests/AppBundle/Controller/AdherentControllerTest.php]: I create a PR to fix typo. Can you please change to 

> functional

**Pull request merged**